---
title: "MICR575_FinalProj"
format: gfm
editor: visual
---

## Goals

The goal of this final project was to process RNA seq data using a workflow management system on a high performance cluster (UTK - ISAAC).

## Main Elements

-   Working remotely on HPC (ISAAC

-   Workflow management system: Snakemake

-   Use of a package manager (mamba)

-   Document w/ R quarto file

-   Version control via GitHub

## Results

1.  worked within the ISAAC System

2.  Processed a small two sample genomic data set with Snakemake for variant calling (inlcuded mamba as package manager)

    -   Snakefile code:

        ``` python
        SAMPLES = ["A", "B"]

        rule all:
            input:
                "plots/quals.svg"

        rule bwa_map:
            input:
                "data/genome.fa",
                "data/samples/{sample}.fastq"
            output:
                "mapped_reads/{sample}.bam"
            shell:
                "bwa mem {input} | samtools view -Sb - > {output}"

        rule samtools_sort:
            input:
                "mapped_reads/{sample}.bam"
            output:
                "sorted_reads/{sample}.bam"
            shell:
                "samtools sort -T sorted_reads/{wildcards.sample} "
                "-O bam {input} > {output}"

        rule samtools_index:
            input:
                "sorted_reads/{sample}.bam"
            output:
                "sorted_reads/{sample}.bam.bai"
            shell:
                "samtools index {input}"

        rule bcftools_call:
            input:
                fa="data/genome.fa",
                bam=expand("sorted_reads/{sample}.bam", sample=SAMPLES),
                bai=expand("sorted_reads/{sample}.bam.bai", sample=SAMPLES)
            output:
                "calls/all.vcf"
            shell:
                "bcftools mpileup -f {input.fa} {input.bam} | "
                "bcftools call -mv - > {output}"

        rule plot_quals:
            input:
                "calls/all.vcf"
            output:
                "plots/quals.svg"
            script:
                "scripts/plot-quals.py"

        ```

    -   environment.yaml code:

        ``` python
        channels:
          - bioconda
          - conda-forge
        dependencies:
          - snakemake-minimal >=5.24.1
          - jinja2 =2.11
          - networkx =2.5
          - matplotlib =3.3
          - graphviz =2.42
          - bcftools =1.9
          - samtools =1.9
          - bwa =0.7
          - pysam =0.15
        ```

    -   Main output files: DAG file of project

        ![](tutorial_material/dag.svg)

    ```{=html}
    <!-- -->
    ```
    -   Main output files: quality file of processed reads

        ![](tutorial_material/quals.svg)

3.  Uploaded files and process to github

## Shortcomings

1.  Unable to get slurm scheduling code worked out
2.  RNA seq data was not processed, just test small genomics data set.
